docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/upgrade-guides/6-0-0-upgrade-guide/index.md

```diff
@@ -0,0 +1,153 @@
+---
+title: "6.0.0 upgrade guide"
+sidebar_position: -20
+---
+
+```mdx-code-block
+import License from "@site/docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/reusable/license/_index.mdx"
+
+<License/>
+```
+
+## Redshift invalid schema evolution recovery
+
+### What is schema evolution?
+
+One of Snowplow’s key features is the ability to [define custom schemas and validate events against them](/docs/understanding-your-pipeline/schemas/). Over time, users often evolve the schemas, e.g. by adding new fields or changing existing fields. To accommodate these changes, RDB loader automatically adjusts the database tables in the warehouse accordingly.
+
+There are two main types of schema changes:
+
+**Breaking**: The schema version has to be changed in a major way (`1-2-3` → `2-0-0`). In Redshift, each major schema version has its own table (`..._1`, `..._2`, etc, for example: `com_snowplowanalytics_snowplow_ad_click_1`).
+
+**Non-breaking**: The schema version can be changed in a minor way (`1-2-3` → `1-3-0` or `1-2-3` → `1-2-4`). Data is stored in the same database table.
+
+### How it used to work
+
+In the past, the transformer would format the data according to the latest version of the schema it saw (for a given major version, e.g. `1-*-*`). For example, if a batch contained events with schema versions `1-0-0`, `1-0-1` and `1-0-2`, the transformer would structure the shredded data based on version `1-0-2`. Then the loader would adjust the database table and load the file.
```

Comment:
I don't think this is an exact picture of how it worked so far, especially in case iglu server has 1-0-3 or later versions

pre-6.0.0, for an entity, transformer would fetch all schemas (which conforms to the `vendor/name/model-*-*` criterion of the entity) from iglu server, merge them, extract properties from json, stringify using tab char as delimeter and put null char in case the property is missing

pre-6.0.0, loader would do the same and this could lead to a race condition hence loading failure as explained [here](https://github.com/snowplow/engineering-announcements/pull/103/files#diff-806b4b8b31cd7d65fa9755b01277baddc7eb35772cc7140331f4060f767080f0R9-R11)

As of 6.0.0, loader migrates up to the max SchemaVer seen in the shredding_complete batch, not in the iglu server